In [1]:
import pandas as pd

In [3]:
data = pd.read_csv("HW1-data.csv")

In [4]:
data.head()

User  Gender (1 =F, 0=M)  260: Star Wars: Episode IV - A New Hope (1977)  \
0   755                   0                                             1.0   
1  5277                   0                                             5.0   
2  1577                   1                                             NaN   
3  4388                   0                                             NaN   
4  1202                   1                                             4.0   

   1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
0                                                5.0         
1                                                3.0         
2                                                NaN         
3                                                3.0         
4                                                3.0         

   356: Forrest Gump (1994)  318: Shawshank Redemption, The (1994)  \
0                       2.0                                    NaN   
1                       NaN                                    2.0   
2                       NaN                                    5.0   
3                       NaN                                    NaN   
4                       4.0                                    1.0   

   593: Silence of the Lambs, The (1991)  3578: Gladiator (2000)  \
0                                    4.0                     4.0   
1                                    4.0                     2.0   
2                                    2.0                     NaN   
3                                    NaN                     1.0   
4                                    4.0                     1.0   

   1: Toy Story (1995)  2028: Saving Private Ryan (1998)       ...         \
0                  2.0                               2.0       ...          
1                  1.0                               NaN       ...          
2                  4.0                               NaN       ...          
3                  2.0                               3.0       ...          
4                  NaN                               4.0       ...          

   2396: Shakespeare in Love (1998)  2916: Total Recall (1990)  \
0                               2.0                        NaN   
1                               3.0                        2.0   
2                               NaN                        1.0   
3                               NaN                        4.0   
4                               5.0                        1.0   

   780: Independence Day (ID4) (1996)  541: Blade Runner (1982)  \
0                                 5.0                       2.0   
1                                 2.0                       NaN   
2                                 4.0                       4.0   
3                                 1.0                       3.0   
4                                 NaN                       4.0   

   1265: Groundhog Day (1993)  2571: Matrix, The (1999)  \
0                         NaN                       4.0   
1                         2.0                       NaN   
2                         1.0                       1.0   
3                         5.0                       NaN   
4                         NaN                       3.0   

   527: Schindler's List (1993)  2762: Sixth Sense, The (1999)  \
0                           2.0                            5.0   
1                           5.0                            1.0   
2                           2.0                            3.0   
3                           5.0                            1.0   
4                           5.0                            5.0   

   1198: Raiders of the Lost Ark (1981)  34: Babe (1995)  
0                                   NaN              NaN  
1                                   3.0              NaN  
2                                   1.0              3.0  
3                                   1.0              2.0  
4              

In [7]:
data.mean()

User                                                       3658.100000
Gender (1 =F, 0=M)                                            0.450000
260: Star Wars: Episode IV - A New Hope (1977)                3.266667
1210: Star Wars: Episode VI - Return of the Jedi (1983)       3.000000
356: Forrest Gump (1994)                                      2.700000
318: Shawshank Redemption, The (1994)                         3.600000
593: Silence of the Lambs, The (1991)                         3.062500
3578: Gladiator (2000)                                        2.916667
1: Toy Story (1995)                                           2.823529
2028: Saving Private Ryan (1998)                              3.000000
296: Pulp Fiction (1994)                                      3.000000
1259: Stand by Me (1986)                                      2.636364
2396: Shakespeare in Love (1998)                              2.909091
2916: Total Recall (1990)                                     1.916667
780: I

Mean Rating: Calculate the mean rating for each movie, order with 
the highest rating listed first, and 

submit the top three : 
1. 318: Shawshank Redemption, The (1994)                         3.600000
2. 260: Star Wars: Episode IV - A New Hope (1977)                3.266667
3. 541: Blade Runner (1982)                                      3.222222

(along with the mean scores for the top two).

In [10]:
data.count()

User                                                       20
Gender (1 =F, 0=M)                                         20
260: Star Wars: Episode IV - A New Hope (1977)             15
1210: Star Wars: Episode VI - Return of the Jedi (1983)    14
356: Forrest Gump (1994)                                   10
318: Shawshank Redemption, The (1994)                      10
593: Silence of the Lambs, The (1991)                      16
3578: Gladiator (2000)                                     12
1: Toy Story (1995)                                        17
2028: Saving Private Ryan (1998)                           11
296: Pulp Fiction (1994)                                   11
1259: Stand by Me (1986)                                   11
2396: Shakespeare in Love (1998)                           11
2916: Total Recall (1990)                                  12
780: Independence Day (ID4) (1996)                         13
541: Blade Runner (1982)                                    9
1265: Gr

1. 1: Toy Story (1995)                                        17
2. 593: Silence of the Lambs, The (1991)                      16
3. 260: Star Wars: Episode IV - A New Hope (1977)             15

In [11]:
df = data.drop(columns=['User','Gender (1 =F, 0=M)'])

In [18]:
rat4higher = df[df>3]

In [23]:
(rat4higher.count()/df.count())*100

260: Star Wars: Episode IV - A New Hope (1977)             53.333333
1210: Star Wars: Episode VI - Return of the Jedi (1983)    35.714286
356: Forrest Gump (1994)                                   30.000000
318: Shawshank Redemption, The (1994)                      70.000000
593: Silence of the Lambs, The (1991)                      43.750000
3578: Gladiator (2000)                                     50.000000
1: Toy Story (1995)                                        35.294118
2028: Saving Private Ryan (1998)                           36.363636
296: Pulp Fiction (1994)                                   36.363636
1259: Stand by Me (1986)                                   36.363636
2396: Shakespeare in Love (1998)                           27.272727
2916: Total Recall (1990)                                   8.333333
780: Independence Day (ID4) (1996)                         30.769231
541: Blade Runner (1982)                                   44.444444
1265: Groundhog Day (1993)        

1. 318: Shawshank Redemption, The (1994)                      70.000000
2. 260: Star Wars: Episode IV - A New Hope (1977)             53.333333
3. 3578: Gladiator (2000)                                     50.000000

In [24]:
df

260: Star Wars: Episode IV - A New Hope (1977)  \
0                                              1.0   
1                                              5.0   
2                                              NaN   
3                                              NaN   
4                                              4.0   
5                                              2.0   
6                                              NaN   
7                                              4.0   
8                                              5.0   
9                                              4.0   
10                                             2.0   
11                                             3.0   
12                                             2.0   
13                                             3.0   
14                                             4.0   
15                                             NaN   
16                                             NaN   
17                                             4.0   
18                                             5.0   
19                                             1.0   

    1210: Star Wars: Episode VI - Return of the Jedi (1983)  \
0                                                 5.0         
1                                                 3.0         
2                                                 NaN         
3                                                 3.0         
4                                                 3.0         
5                                                 4.0         
6                                                 NaN         
7                                                 NaN         
8                                                 1.0         
9                                                 2.0         
10                                                1.0         
11                                                5.0         
12                                                3.0         
13                                                NaN         
14                                                4.0         
15                                                NaN         
16                                                NaN         
17                                                5.0         
18                                                1.0         
19                                                2.0         

    356: Forrest Gump (1994)  318: Shawshank Redemption, The (1994)  \
0                        2.0                                    NaN   
1                        NaN                                    2.0   
2                        NaN                                    5.0   
3                        NaN                                    NaN   
4                        4.0                                    1.0   
5                        4.0                                    4.0   
6                        3.0                                    1.0   
7                        NaN                                    NaN   
8                        NaN                                    4.0   
9                        NaN                                    5.0   
10                       5.0                                    NaN   
11                       2.0                                    NaN   
12                       NaN                                    5.0   
13                       3.0                                    NaN   
14                       NaN                                    NaN   
15                       1.0                                    NaN   
16                       NaN                                    NaN   
17                       1.0                                    5.0   
18                       NaN                                    NaN   
19                       2.0                                    4.0   

    593: Silence of the Lambs, The (1991)  3578: Glad